dtfs-central-api/src/v1/validation/utilisation-report-service/selected-fee-record-validator.test.ts

```diff
@@ -0,0 +1,47 @@
+import { Currency, FeeRecordEntity, FeeRecordEntityMockBuilder, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { validateSelectedFeeRecordsAllHaveSamePaymentCurrency } from './selected-fee-record-validator';
+import { InvalidPayloadError } from '../../../errors';
+
+describe('selected fee record validator', () => {
+  describe('validateSelectedFeeRecordsAllHaveSamePaymentCurrency', () => {
+    it('does not throw if only one fee record', () => {
+      // Arrange
+      const selectedFeeRecords = [
+        FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build()).build(),
+      ];
+
+      // Act + Assert
+      expect(() => validateSelectedFeeRecordsAllHaveSamePaymentCurrency(selectedFeeRecords)).not.toThrow();
+    });
+
+    it('does not throw if no fee records provided', () => {
+      // Arrange
+      const selectedFeeRecords: FeeRecordEntity[] = [];
+
+      // Act + Assert
+      expect(() => validateSelectedFeeRecordsAllHaveSamePaymentCurrency(selectedFeeRecords)).not.toThrow();
+    });
+
+    it('does not throw if no all fee records have the same payment currency', () => {
+      // Arrange
+      const selectedFeeRecords = [aFeeRecordWithPaymentCurrency('GBP'), aFeeRecordWithPaymentCurrency('GBP')];
+
+      // Act + Assert
+      expect(() => validateSelectedFeeRecordsAllHaveSamePaymentCurrency(selectedFeeRecords)).not.toThrow();
+    });
+
+    it('throws invalid payload error if any two fee records have differing payment currency', () => {
+      // Arrange
+      const selectedFeeRecords = [aFeeRecordWithPaymentCurrency('GBP'), aFeeRecordWithPaymentCurrency('JPY')];
+
+      // Act + Assert
+      expect(() => validateSelectedFeeRecordsAllHaveSamePaymentCurrency(selectedFeeRecords)).toThrow(InvalidPayloadError);
+    });
+  });
+
+  function aFeeRecordWithPaymentCurrency(paymentCurrency: Currency) {
+    return FeeRecordEntityMockBuilder.forReport(UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS').build())
+      .withPaymentCurrency(paymentCurrency)
+      .build();
+  }
```

Comment:
Could this be moved to the top and could we use an arrow function for it?

Reply:
Replied to in another thread: it's kept at the bottom as the way this function works isn't really important for the test